src/thread/thread.cpp

```diff
@@ -156,6 +156,17 @@ namespace fc {
 
    void          thread::debug( const fc::string& d ) { /*my->debug(d);*/ }
 
+#if defined(__linux__) || defined(__APPLE__)
+#include <signal.h>
+#endif
+
+   void thread::signal(int sig)
+   {
+#if defined(__linux__) || defined(__APPLE__)
+      pthread_kill( my->boost_thread->native_handle(), sig );
```

Comment:
Hm. Bypassing the `boost::thread` abstraction and using the underlying thread implementation's signal mechanism is a bit dirty. I haven't found a portable way to do this with boost::thread directly, though.

Reply:
I guess usually we don't do inter thread communication with IPC signals, perhaps that's why `boost::thread` doesn't have it. This time I used IPC signals because it's the only way I have found to interrupt the blocking `read()` in *editline* which is C-style. Perhaps we can change that `read()` to something else E.G. `select()` or `poll()`, but it would need much bigger efforts. So IMHO it's fine to take the quick and dirty approach.